ifsbench/config_mixin.py

```diff
@@ -39,67 +48,133 @@ def from_config(
         Returns:
             class instance
         """
-        raise NotImplementedError()
+        return cls(**config)
 
-    @abstractmethod
     def dump_config(
         self, with_class: bool = False
-    ) -> Dict[str, Union[str, float, int, bool, List]]:
+    ) -> Dict[str, Union[str, float, int, bool, List, None]]:
         """Get configuration for output.
 
         Args:
-            with_class: Add CLASSNAME key with class name to configuration.
+            with_class: Add/keep CLASSNAME key with class name to configuration.
 
         Returns:
             Configuration that can be used to create instance.
         """
-        raise NotImplementedError()
+        config = self.model_dump(exclude_none=True, round_trip=True)
+
+        # Manually convert Path objects to str. In theorey, the subsequent
+        # TypeAliasType thing should be able to do this but for some reason
+        # this doesn't work.
+        for k, v in config.items():
+            if isinstance(v, Path):
+                config[k] = str(v)
+
+        # Add class name to the dictionary (or remove it if with_class==False).
+        if with_class:
+            config[CLASSNAME] = type(self).__name__
+        else:
+            config.pop(CLASSNAME, None)
 
+        # Make sure that the output is indeed only dict/list/str/int/float/bool/None.
+        # To do this, we use the pydantic validation. First, we define this recursive
+        # data type.
+        Allowed = TypeAliasType(
+            'Allowed',
+            'Union[Dict[Allowed, Allowed], List[Allowed], str, int, float, bool, None]',
+        )
 
-class PydanticConfigMixin(ConfigMixin, BaseModel, use_enum_values=True):
+        allowed_type = TypeAdapter(Dict[str, Allowed])
+
+        return allowed_type.validate_python(config)
+
+class AbstractDataClass(DataClass):
```

Comment:
Sorry to harp on the names so much, but I find this rather confusing: usually, you'd have an AbstractDataClass and then the implemented DataClass. Here, it's the other way round. Maybe call this SubclassableDataClass

Reply:
So this is now SerialisationMixin, but it still has this unexpected inversion of which one is Abstract. Could you still rename the AbstractSerialisationMixin to something that indicates which part is abstract and that it is a type if SerialisationMixin, not the other way round?